cmd/up/up.go

```diff
@@ -258,11 +259,15 @@ func (up *upContext) activate(autoDeploy, build bool) {
 	isRetry := false
 
 	for {
+		// create a new context on every iteration
+		ctx, cancel := context.WithCancel(context.Background())
```

Comment:
this is the main change. we need to create the context on every iteration, since it is canceled at this level when there's an error and we retry.